### PR TITLE
Azure mirror close with closing reason already set

### PR DESCRIPTION
**Pull request by DinaMeylakh**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-hq.paloaltonetworks.local/browse/XSUP-28047)

## Description
Fixed an issue where providing a close reason separately before closing the incident in XSOAR sometimes resulted in the closed status not being mirrored to Microsoft Sentinel.

## Must have
- [x] Tests
- [ ] Documentation

### CODE REVIEW

**Comment by DinaMeylakh** [comment]
TDD: failing unittests commit [1551974](https://github.com/demisto/content/pull/30291/commits/1551974e64cb4f476c6891a5a03da0cc690b822b) in [pipeline](https://code.pan.run/xsoar/content/-/jobs/34880600)

File: Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.py [hunk 1]
```diff
@@ -648,17 +648,19 @@ def get_mapping_fields_command() -> GetMappingFieldsResponse:
...
+        2. The incident current status is not closed.
+        3. A closing reason was provided (either in the delta or before in the data).
 
     The second is mandatory, so a closing request will not be sent for all mirroring requests that occur after closing an incident
     (in the case where the incident has been updated, but the status has not been changed).
     """
     closing_field = 'classification'
-    return demisto.params().get('close_ticket') and closing_field in delta
+    closing_reason = delta.get(closing_field, data.get(closing_field, ''))
+    return demisto.params().get('close_ticket') and data.get('status', '') != 'Closed' and bool(closing_reason)
```
Comment (ShahafBenYakir) [inline Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.py:663]: We've added a new condition `data.get('status', '') != 'Closed'` why is it necessary?

**Review by ShahafBenYakir — COMMENTED** [review]
Great work

File: Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.py (hunk 1 above)
Comment (DinaMeylakh) [reply Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.py:663]: So we won't send a closing request to an already closed ticket (every time any other field is updated). If I understood the function documentation correctly, this is why the closing field was checked in the delta in the first place.

**Review by ShahafBenYakir — APPROVED** [review]

**Pull request merged**